arbitrum-client/integration/main.rs

```diff
@@ -0,0 +1,95 @@
+#![deny(unsafe_code)]
+#![deny(missing_docs)]
+#![deny(clippy::missing_docs_in_private_items)]
+#![allow(incomplete_features)]
+#![feature(generic_const_exprs)]
+
+mod constants;
+mod helpers;
+
+use arbitrum_client::{
+    client::{ArbitrumClient, ArbitrumClientConfig},
+    constants::Chain,
+};
+use clap::Parser;
+use constants::{DARKPOOL_CONTRACT_KEY, DARKPOOL_PROXY_CONTRACT_KEY};
+use helpers::parse_addr_from_deployments_file;
+use test_helpers::integration_test_main;
+use util::{logging::LevelFilter, runtime::block_on_result};
+
+use crate::constants::{DEFAULT_DEVNET_HOSTPORT, DEFAULT_DEVNET_PKEY};
+
+#[derive(Debug, Clone, Parser)]
+#[command(author, version, about, long_about=None)]
+struct CliArgs {
+    /// The private key to use for signing transactions in the integration test
+    ///
+    /// Defaults to the private key that the Nitro devnet is pre-seeded with
+    #[arg(short = 'p', long, default_value = DEFAULT_DEVNET_PKEY)]
+    private_key: String,
+
+    /// The location of a `deployments.json` file that contains the addresses of
+    /// the deployed contracts
+    #[arg(short, long)]
+    deployments_path: String,
+
+    // TODO: Add a flag for the contract artifacts to allow for building/deploying
+    // during test setup
+    /// The url of the Arbitrum RPC endpoint to use for the integration test
+    #[arg(long, default_value = DEFAULT_DEVNET_HOSTPORT)]
+    rpc_url: String,
+
+    /// The test to run
+    #[arg(short, long, value_parser)]
+    test: Option<String>,
+
+    /// The verbosity level of the test harness
+    #[arg(short, long)]
+    verbose: bool,
+}
+
+/// The arguments provided to every integration test
+#[derive(Clone)]
+struct IntegrationTestArgs {
+    /// The Arbitrum client that resolves to a locally running devnet node
+    arbitrum_client: ArbitrumClient,
```

Comment:
nit: just do `client`, this is a bit verbose

Reply:
"drop the arbitrum, it's cleaner"